code/modules/cm_aliens/structures/special/pylon_core.dm

```diff
@@ -178,22 +208,91 @@
 
 	activated = TRUE
 	linked_hive.check_if_hit_larva_from_pylon_limit()
-	addtimer(CALLBACK(src, PROC_REF(give_larva)), XENO_PYLON_ACTIVATION_COOLDOWN, TIMER_UNIQUE|TIMER_OVERRIDE|TIMER_LOOP|TIMER_DELETE_ME)
+	addtimer(CALLBACK(src, PROC_REF(give_royal_resin)), XENO_PYLON_ACTIVATION_COOLDOWN, TIMER_UNIQUE|TIMER_OVERRIDE|TIMER_LOOP|TIMER_DELETE_ME)
 
 /// Looped proc via timer to give larva after time
-/obj/effect/alien/resin/special/pylon/endgame/proc/give_larva()
+/obj/effect/alien/resin/special/pylon/endgame/proc/give_royal_resin()
 	if(!activated)
 		return
 
 	if(!linked_hive.hive_location || !linked_hive.living_xeno_queen)
 		return
 
-	if(linked_hive.check_if_hit_larva_from_pylon_limit())
```

Comment:
If you still want `hit_larva_pylon_limit` to be accurate, we need to still call that proc here since this was the only looping place that updates it.